include/dg/analysis/PointsTo/PointerSubgraphValidator.h

```diff
@@ -26,8 +27,8 @@ class PointerSubgraphValidator {
     bool no_connectivity;
 
 protected:
-    std::string errors{};
-    std::string warnings{};
+    std::stringstream errors;
+    std::stringstream warnings;
```

Comment:
It would be actually better to use a generic `ostream`, so that we can just forward the errors or warnings to stderr (we can get rid of the `getErrors` and `getWarnings` methods then).

Reply:
I see, that would be better. The pointer validator is used with llvm output streams, but I don't want to use `llvm::raw_ostream` in generic part of dg. I have two ideas how to resolve this: 
1. remember `std::ostream&` in the validator and also in the llvm module
2. turn pta validator into a template class parameterized by the type of the stream and remember references to stream

Do you have any other ideas how to solve it? Number 2 seems a bit cleaner to me, but it's more work so I thought it's better to discuss this first.

Reply:
I think that 1 is good. The design of the validator is a bit clumsy and turning it into a template would drive one mad :)